src/Hacks/autoblock.cpp

```diff
@@ -43,13 +42,35 @@ void Autoblock::CreateMove(CUserCmd* cmd)
 	if (!target)
 		return;
 
+	bool crouchBlock = false;
+	if (target->GetBonePosition(BONE_PELVIS).z < localplayer->GetAbsOrigin().z &&
+		Vector(localplayer->GetAbsOrigin() - target->GetAbsOrigin()).LengthSqr() < 1e5)
```

Comment:
If you could do something like
`localplayer->GetAbsOrigin().z - target->GetAbsOrigin().z >= entity->GetCollideable()->OBBMaxs().z`
and add other checks that i'm on top then i **_THINK_** it would be working better.